app/src/main/java/com/github/se/polyfit/ui/screen/recipeRec/RecipeRecommendationScreen.kt

```diff
@@ -58,15 +62,31 @@ fun recipeDisplay(
               fontWeight = FontWeight.Bold,
           )
         }
+        if (recipes.value.isEmpty()) {
+          item { loader(paddingValues) }
+          return@LazyColumn
+        }
+
         recipes.value.forEach {
           item {
             RecipeCard(
                 recipe = it,
                 onCardClick = {
-                  (recipesRec::setRecipe)(it)
+                  (recipesRec::onSelectedRecipe)(it)
                   navigateToRecipeRecommendationMore.invoke()
                 })
           }
         }
       }
 }
+
+@Composable
+fun loader(paddingValues: PaddingValues) {
+
+  Box(
+      modifier = Modifier.fillMaxSize().padding(paddingValues).testTag("Loader"),
+      contentAlignment = Alignment.Center) {
+        CircularProgressIndicator(modifier = Modifier.padding(16.dp).testTag("LoadingPost"))
+      }
+  return
+}
```

Comment:
```suggestion
@Composable
fun Loader(paddingValues: PaddingValues) {
  Box(
      modifier = Modifier.fillMaxSize().padding(paddingValues).testTag("Loader"),
      contentAlignment = Alignment.Center) {
        CircularProgressIndicator(modifier = Modifier.padding(16.dp).testTag("LoadingPost"))
      }
}
```
No need to return early here (unless I'm missing something?). Also i think general rule of thumb for composable functions is Title case (all first letter caps). Same for line 43 in this file